third_party/nvfuser/csrc/python_frontend/fusion_record.h

```diff
@@ -64,7 +66,7 @@ struct RecordFunctor {
       std::vector<State> _args,
       std::vector<State> _outputs,
       std::string _name,
-      RecordType _record_type)
+      serde::RecordType _record_type)
```

Comment:
Maybe I wasn't paying enough attention.

looks like we now have a single RecordType for each op, I'm wondering if the op/name is still useful to be passed in separately here.

Reply:
The `RecordType` corresponds to the `RecordFunctor` child class and its template arguments. The `name` string represents the specific operation.

e.g., For the `abs` unary op, its record functor is `OpRecord<Tensorview*, TensorView*>`. Its `RecordType` is `Unary_TV` and its name is `abs`.

Reply:
So it sounds like that part is still following Kevin's old design. What's the motivation of differentiation reduction ops with
`serde::RecordType::RecordType_ReductionMax/Min/Sum`? Seems like we are still passing functor to handle those.

Reply:
In [PR 2516](https://github.com/csarofeen/pytorch/pull/2516/files#diff-e1f150898fcfedeeb5caf66cca2550b557fece8881c0b7fd8f648fda4dd35556R76-R85), I use a `string` to `std::function` map to get the correct NvFuser function.

Since there are only three reduction operations, it seemed inefficient to use the map approach. I defined them explicitly in the `RecordType` enum.